src/app/modules/admin-shared/dynamic-form/dynamic-form.component.ts

```diff
@@ -1,29 +1,30 @@
-import { Component, Input, OnInit }  from '@angular/core';
-import { FormGroup }                 from '@angular/forms';
-import { QuestionBase } from '../form/question-base';
+import { Component, Input, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
 import { UsersService } from '../../admin-core';
-// import { QuestionBase }              from './question-base';
-// import { QuestionControlService }    from './question-control.service';
-UsersService
+import { SingleUserComponent } from '../../users/single-user/single-user.component';
+
 
 @Component({
   selector: 'app-dynamic-form',
   templateUrl: './dynamic-form.component.html',
-  providers: [ UsersService ]
+  providers: [UsersService]
 })
 export class DynamicFormComponent implements OnInit {
 
-  @Input() questions: QuestionBase<string>[] = [];
+  @Input() questions: any;
   form: FormGroup;
   payLoad = '';
 
-  constructor(private qcs: UsersService) {  }
+  constructor(private qcs: UsersService, private singleuser: SingleUserComponent) {
+  }
 
   ngOnInit() {
     this.form = this.qcs.toFormGroup(this.questions);
+    console.log('===========', this.form);
   }
 
   onSubmit() {
-    this.payLoad = JSON.stringify(this.form.getRawValue());
+    this.payLoad = this.form.getRawValue();
+    this.singleuser.createUser(this.payLoad);
```

Comment:
Purpose of dynamic form is to be reusable for all the forms..

Why are you hardcoding for single user page?

Reply:
removed